### ISSUE TITLE
`chakra-ui-color-mode` cookie not being set in Safari, causing page flash on load and issues with `useColorModeValue`

### ISSUE THREAD

**Issue by SpencerKaiser** [chakra-ui/chakra-ui#5641]
### Description

chakra-ui-color-mode not being set in Safari, causing page flash on load

### Link to Reproduction

https://codesandbox.io/s/jolly-ramanujan-uz3t95?file=/src/index.tsx

### Steps to reproduce

1. Follow steps to add Chakra to an existing React project (Next is what I used)
2. Build and start the app
3. Refresh page from origin
4. Observe page flash and check `local storage` for `chakra-ui-color-mode` cookie, which will not be set

### Chakra UI Version

1.8.5

### Browser

Safari Version 15.2 (17612.3.6.1.6)

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

The cookie is properly set in Chrome, so this issue only takes place in Safari.

The issue also impacts the proper color rendering for components at the `page` level vs within a component file; the former will use the wrong color value unless the cookie is set.

I tried reproducing making a fully reproducible one in codesandbox, but it looks like the `next` one is fragile and breaks once you create a single error; after 6 attempts, I gave up and just made one locally. [Here's the repo](https://github.com/SpencerKaiser/flash-repro) and below is a screenshot of the example I created within it.

<img width="965" alt="image" src="https://user-images.githubusercontent.com/6445731/155004882-d261bfcd-ac1d-4c8f-a05c-a4f7298aa394.png">

**Comment by kaityhallman** [comment]
I'm also seeing this issue in chrome. The local storage value is not being set.

**Comment by CloudLooks** [comment]
Reporting a similar issue in Safari, maybe related. While toggling colorMode, the data-theme value on `html` remains unchanged but the body class does update from "chakra-ui-light" to "chakra-ui-dark"

**Comment by CloudLooks** [comment]
To follow up, my workaround was to apply a bg color to each Page in my app, and not rely on `body` for the backgroundColor. not ideal but it is working

**Comment by SpencerKaiser** [comment]
Not stale 😅

**Comment by akroll1** [comment]
I also had this issue and it bothered me. In my debugging I couldn't get the initialColorMode prop to work in the ColorModeScript, which was pretty confusing for awhile. I put a config in a custom theme I created and it is respecting that.

I think another thing to watch out for that the script seems to get cached there, I would clear out the local storage flag for this and open a new tab... Maybe not, I forget all the steps I took lol but I know the initialColorMode prop is not working in ^2.

**Comment by SpencerKaiser** [comment]
Not stale 😅 (part 2)

**Comment by SpencerKaiser** [comment]
Not stale? (bot didn't trigger)

**Comment by NBMSacha** [comment]
Not stale.. :/

**Comment by SpencerKaiser** [comment]
@stale bot has very selective hearing 😓 

Not stale? 🙃

**Comment by SpencerKaiser** [comment]
But also fwiw, @NBMSacha this issue seems to be resolved in Chakra 2.x... it was quit the journey getting everything to work (`useStandaloneToast` was quite annoying since there were major breaking changes there), but now the cookie seems to be set/used correctly everywhere!

Also... not stale! (bot _**still**_ didn't trigger)

**Comment by SpencerKaiser** [comment]
Just throwing a comment here to keep Stale-bot away 👎 🤖

**Comment by NBMSacha** [comment]
Oh, really? Without anything more to do than just to update to version 2? It will automatically be fixed?

**Comment by SpencerKaiser** [comment]
Yep! The value is being set now, but there are other problems with the color mode stuff (#6790, for example). Honestly this one might be closable - you should upgrade and see if it cleared up for you.
